### PR TITLE
Profiles for gramps, newsboat and freeoffice

### PR DESCRIPTION
**Pull request by curiosity-seeker**
These profiles work well for me on Manjaro KDE.

### CODE REVIEW

File: etc/newsboat.profile [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+# private
+# private-bin newsboat bash
+private-cache
+private-dev
+# private-etc none
+#private-lib
+private-tmp
+
+memory-deny-write-execute
+#noexec ${HOME}
```
Comment (rusty-snake) [inline etc/newsboat.profile]: Remove `noexec` because this is in `disable-exec`

File: etc/newsboat.profile [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+notv
+nou2f
+novideo
+protocol inet,inet6
+seccomp
+shell none
+
+disable-mnt
+# private
+# private-bin newsboat bash
```
Comment (rusty-snake) [inline etc/newsboat.profile]: no whitespace, `newsboat,bash`

File: etc/newsboat.profile [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+# nosound
+notv
+nou2f
+novideo
+protocol inet,inet6
+seccomp
+shell none
+
+disable-mnt
+# private
```
Comment (rusty-snake) [inline etc/newsboat.profile]: remove

File: etc/gramps.profile [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+nosound
+notv
+nou2f
+novideo
+protocol unix,inet,inet6
+seccomp
+shell none
+
+disable-mnt
+# private
```
Comment (rusty-snake) [inline etc/gramps.profile]: remove

File: etc/gramps.profile [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+# private-bin program
```
Comment (rusty-snake) [inline etc/gramps.profile]: remove or use `gramps` instead of program, this doesn't make sense

File: etc/gramps.profile [hunk 6]
```diff
@@ -0,0 +1,50 @@
+# Firejail profile for gramps
+# Description: genealogy program
+# This file is overwritten after every install/update
+# Persistent local customizations
+include gramps.local
+# Persistent global definitions
+include globals.local
+
+noblacklist ${PATH}/python3*
```
Comment (rusty-snake) [inline etc/gramps.profile]:
```
# Allow python (blacklisted by disable-interpreters.inc)
#noblacklist ${PATH}/python2*
noblacklist ${PATH}/python3*
#noblacklist /usr/lib/python2*
noblacklist /usr/lib/python3*
#noblacklist /usr/local/lib/python2*
noblacklist /usr/local/lib/python3*
```

File: etc/freeoffice-textmaker [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+nou2f
+novideo
+protocol unix,inet,inet6
+seccomp
+shell none
+tracelog
+
+# disable-mnt
+# private
+# private-bin program
```
Comment (rusty-snake) [inline etc/freeoffice-textmaker]: remove or add all needed programs

File: etc/freeoffice-textmaker [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+include disable-exec.inc
+include disable-interpreters.inc
+include disable-passwdmgr.inc
+include disable-programs.inc
+# include disable-xdg.inc
+
+# apparmor
+caps.drop all
+ipc-namespace
+machine-id
```
Comment (rusty-snake) [inline etc/freeoffice-textmaker]: breaks sound, because of `# nosound` it should be removed

File: etc/freeoffice-textmaker [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+net none
```
Comment (rusty-snake) [inline etc/freeoffice-textmaker]: `net none` or `netfilter`

File: etc/freeoffice-textmaker (hunk 8 above)
Comment (curiosity-seeker) [reply etc/freeoffice-textmaker]: Sorry, I don't understand that remark. Why does it break sound if `nosound` is commented?

All other proposed changes added to the profiles.

Comment (rusty-snake) [reply etc/freeoffice-textmaker]: The why is a little difficult, but it is a fact that machine-id breaks the sound (If pulse is used).

Comment (curiosity-seeker) [reply etc/freeoffice-textmaker]: Okay, removed.

**Comment by rusty-snake** [comment]
I played a littlebit with the newsboat.profile and come to [this](https://gist.github.com/rusty-snake/8652584e91c4ef1de3e7b531e45de2e2).

File: etc/freeoffice-planmaker [hunk 10]
```diff
@@ -0,0 +1,45 @@
+# Firejail profile for freeoffice-planmaker
```
Comment (SkewedZeppelin) [inline etc/freeoffice-planmaker]: no extension?
append .profile

Comment (SkewedZeppelin) [inline etc/freeoffice-planmaker]: no extension?
append .profile

File: etc/freeoffice-presentations [hunk 11]
```diff
@@ -0,0 +1,45 @@
+# Firejail profile for freeoffice-presentations
```
Comment (SkewedZeppelin) [inline etc/freeoffice-presentations]: no extension?
append .profile

File: etc/freeoffice-textmaker [hunk 12]
```diff
@@ -0,0 +1,45 @@
+# Firejail profile for freeoffice-textmaker
```
Comment (SkewedZeppelin) [inline etc/freeoffice-textmaker]: no extension?
append .profile

File: etc/gramps.profile [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+# Allow python (blacklisted by disable-interpreters.inc)
+#noblacklist ${PATH}/python2*
+noblacklist ${PATH}/python3*
+#noblacklist /usr/lib/python2*
+noblacklist /usr/lib/python3*
+#noblacklist /usr/local/lib/python2*
+noblacklist /usr/local/lib/python3*
+
+noblacklist ${HOME}/.gramps
+whitelist ${HOME}/.gramps
```
Comment (rusty-snake) [inline etc/gramps.profile]: Move and add:
```
include disable-xdg.inc

mkdir ${HOME}/.gramps
whitelist ${HOME}/.gramps
include whitelist-common.inc
include whitelist-var-common.inc

apparmor
```

Note: When build a whitelisting-profile you can't store files in dirs like Documents, need gramps this?

Comment (rusty-snake) [reply etc/gramps.profile]: @curiosity-seeker I didn't say something about the noblacklist line ;)
leave it

File: etc/newsboat.profile [hunk 14]
```diff
@@ -0,0 +1,46 @@
+# Firejail profile for Newsboat
+# Description: RSS program
+# This file is overwritten after every install/update
+# Persistent local customizations
+include newsboat.local
+# Persistent global definitions
+include globals.local
+
+noblacklist ${HOME}/.newsboat
+whitelist ${HOME}/.newsboat
```
Comment (rusty-snake) [inline etc/newsboat.profile]: whitelist need mkdir and kann be moved down.

You can also add the other things from [here](https://gist.github.com/rusty-snake/8652584e91c4ef1de3e7b531e45de2e2).

**Comment by rusty-snake** [comment]
Merged, Thanks.

**Pull request merged**